db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -114,14 +114,15 @@ public <A> CompletableFuture<A> runWithArgs(Function<WorkItemArgs, CompletableFu
 
                 if (error instanceof StatusRuntimeException) {
                     StatusRuntimeException ex = (StatusRuntimeException) error;
+                    logger.debug("RunWorkItem[{}] completed with error status: {}", msgId, ex.getStatus());
 
                     if (ex.getStatus().getCode().equals(Status.Code.UNAVAILABLE) || ex.getStatus().getCode().equals(Status.Code.ABORTED)) {
                         self.pushMsg(new CreateChannel(args.id));
                     }
+                } else {
+                    logger.warn("RunWorkItem[{}] completed exceptionally: {}", msgId, error.toString());
                 }
 
-                logger.debug("RunWorkItem[{}] completed exceptionally: {}", msgId, error.getMessage());
```

Comment:
The problem with this statement was that `error.getMessage()` is `null` in many cases, e.g. a StatusRuntimeException with status NOT_FOUND. To better log the more or less expected `StatusRuntimeException` and everything else this statement has been split up into the `if-else` condition.